analytic_engine/src/lib.rs

```diff
@@ -131,3 +138,23 @@ impl Default for ObkvWalConfig {
         }
     }
 }
+
+/// Config of wal based on obkv.
+#[derive(Debug, Default, Clone, Deserialize)]
+#[serde(default)]
+pub struct KafkaWalConfig {
+    /// Kafka client config
+    pub kafka_config: KafkaConfig,
+
+    /// Wal config
+    pub wal_config: MessageQueueWalConfig,
+}
+
+/// Options for storage backend
+#[derive(Debug, Clone, Deserialize)]
+#[serde(tag = "type")]
+pub enum WalStorageConfig {
+    RocksDB,
+    Obkv(Box<ObkvWalConfig>),
```

Comment:
Why use box here?

Reply:
> Why use box here?

It is recommended by `clippy`. 
As I see, maybe because `enum` is C's `union` actually, and one config here is much larger than another.